src/main/java/org/jitsi/jigasi/CallContext.java

```diff
@@ -416,6 +440,15 @@ private void update()
             // target room name
             boshURL = boshURL.replace("{subdomain}", subdomain);
         }
+
+        if (this.authToken != null)
+        {
+            String tokenParam = "&token=" + this.authToken;
+            if (!boshURL.endsWith(tokenParam))
```

Comment:
Can you change it to just contains &token= to minimize string comparison.

Reply:
> Hi, thanks for your contribution!
> If you haven't already done so, could you please make sure you sign our CLA (https://jitsi.org/icla for individuals and https://jitsi.org/ccla for corporations)? We would unfortunately be unable to merge your patch unless we have that piece :(.

under process. will be done shortly.

Reply:
> Can you change it to just contains &token= to minimize string comparison.

wouldn't it break if the BOSH_URL_PATTERN itself contains a "&token=" ? Also, how expensive is the endsWith() comparison? JWT tends to be long, yes, but how often are these comparisons made? update() only after the token is set and the token is set last so basically only once. no?

Reply:
If token is set, don't set it twice, so that sounds ok. Adding two tokens you risk exceeding url size limits as there can be such on some services ...

Reply:
> JWT tends to be long, yes, but how often are these comparisons made? 

Not much, but why doing it, if there is a quicker check.

Reply:
> > JWT tends to be long, yes, but how often are these comparisons made?
> 
> Not much, but why doing it, if there is a quicker check.

done.

Reply:
Hi Damencho,
Do you need anything else to merge this ?
(we modified code according your request and uploaded the CLA)
(we need to package jigasi now and it will be easier if it is from your repos than from our fork) 
Best regards
Sebastien